R/SSplotTags.R

```diff
@@ -43,380 +43,399 @@
+    plotinfo <- rbind(plotinfo, data.frame(file=file, caption=caption))
+    return(plotinfo)
+  }
+  plotinfo <- NULL
 
-    tagdbase2 <- replist[["tagdbase2"]]
-    if (is.null(tagdbase2) || nrow(tagdbase2) == 0) {
-      if (verbose) {
-        message("skipping tag plots because there's no tagging data")
-      }
-    } else {
-      # filter tag groups if requested
-      if (!is.null(taggroups)) {
-        tagdbase2 <- tagdbase2[tagdbase2[["Repl."]] %in% taggroups, ]
-        message(
-          "Filtered tag groups for plotting based on input vector taggroups\n",
-          "Plots will show", length(unique(tagdbase2[["Repl."]])),
-          "out of", length(unique(replist[["tagdbase2"]][["Repl."]])),
-          "total included in the model."
-        )
-      }
+  if(plotdir=="default") plotdir <- replist$inputs$dir
 
-      # calculations needed for printing to multiple PNG files
-      grouprange <- unique(tagdbase2[["Repl."]])
-      ngroups <- length(unique(tagdbase2[["Repl."]]))
-      npages <- ceiling(ngroups / (tagrows * tagcols))
-      nseasons <- replist[["nseasons"]]
-      width <- 0.5 / nseasons
-      tagreportrates <- replist[["tagreportrates"]]
-      tagrelease <- replist[["tagrelease"]]
-      tagsalive <- replist[["tagsalive"]]
-      tagtotrecap <- replist[["tagtotrecap"]]
-      if (is.null(latency)) {
-        latency <- replist[["tagfirstperiod"]]
-      }
+  tagdbase2 <- replist$tagdbase2
+  if(is.null(tagdbase2) || nrow(tagdbase2)==0){
+    if(verbose){
+      message("skipping tag plots because there's no tagging data")
+    }
+  }else{
+    # filter tag groups if requested
+    if(!is.null(taggroups)){
+      tagdbase2 <- tagdbase2[tagdbase2$Rep %in% taggroups,]
+      message("Filtered tag groups for plotting based on input vector taggroups\n",
+              "Plots will show", length(unique(tagdbase2$Rep)),
+              "out of", length(unique(replist$tagdbase2$Rep)),
+              "total included in the model.")
+    }
 
-      tagfun1 <- function(ipage = 0) {
-        # obs & exp recaps by tag group
-        par(
-          mfcol = c(tagrows, tagcols), mar = c(2.5, 2.5, 2, 1),
-          cex.main = cex.main, oma = c(2, 2, 2, 0)
-        )
-        if (npages > 1 & ipage != 0) {
-          grouprange <- intersect(
-            grouprange,
-            1:(tagrows * tagcols) + tagrows * tagcols * (ipage - 1)
-          )
+    # calculations needed for printing to multiple PNG files
+    grouprange     <- unique(tagdbase2$Rep)
+    ngroups        <- length(unique(tagdbase2$Rep))
+    npages         <- ceiling(ngroups/(tagrows*tagcols))
+    nseasons       <- replist$nseasons
+    width          <- 0.5/nseasons
+    tagreportrates <- replist$tagreportrates
+    tagrelease     <- replist$tagrelease
+    tagsalive      <- replist$tagsalive
+    tagtotrecap    <- replist$tagtotrecap
+    if(is.null(latency)){
+      latency <- replist$tagfirstperiod
+    }
+    #Dependencies
+    library(tidyverse)
```

Comment:
If we keep dplyr and ggplot2 functions (rather than bringing in the entire tidyverse), would you please add the packages to the suggested section of the description as mentioned? 

I will go back and add the package name before the function calls. 

Thank you!

Reply:
Included package names before function (see https://github.com/r4ss/r4ss/pull/455/commits/877ac3d3bdb91853fffaad355a8d8e3aea975ff9).